google-cloud-spanner/lib/google/cloud/spanner/convert.rb

```diff
@@ -99,8 +99,7 @@ def object_to_grpc_value obj, field = nil
               if field.is_a? Fields
                 field.struct(obj).to_grpc_value
               else
-                raise ArgumentError,
-                      "A hash value cannot be set to type #{field}."
+                Google::Protobuf::Value.new string_value: obj.to_json
```

Comment:
Just curious: how can we decide to encode a `Hash` to JSON or STRUCT?

Reply:
`Convert.to_input_params_and_types` is used in the `execute_query` to execute SQL query with JSON type  need to specify explicitly `types`. If types are not provided while executing a query with JSON data then it will be converted to STRUCT.

```
client.execute_query "SELECT json from stuffs where JSON_VALUE(json, '$.rating') = JSON_VALUE(@detail, '$.rating')", params: { detail: { rating: 10 } }, types: { detail: :JSON }
```

Reply:
Thanks for the explanation.